### PR TITLE
[Mar-12609] Configure Webhook for nightwatchman in Sendinblue

### PR DESCRIPTION
**Pull request by AnujAngooral**
Added SendInBlue function to handle events coming from Sendinblue for nightwatchman.
Added Test cases to validate

### CODE REVIEW

File: MailReporter/Mvc/Controllers/MailReporterController.cs [hunk 1]
```diff
@@ -82,5 +86,32 @@ public async Task<IActionResult> Mandrill()
             }
             return new OkObjectResult("Event processed successfully");
         }
+
+        [AllowAnonymous]
+        [HttpGet, HttpPost, HttpHead]
+        public async Task<IActionResult> SendInBlue(JObject payload)
+        {
+            //log.Info($"C# HTTP trigger function processed a request. HTTP Method: {req.Method}");
```
Comment (bruno-de-andrade) [inline MailReporter/Mvc/Controllers/MailReporterController.cs]: you can remove all unnecessary comments in the file

**Review by bruno-de-andrade — COMMENTED** [review]
Take care when you commit to always add the ticket number in the commit message, like in the example here in the **Smart Commits** section: https://crossvertise.atlassian.net/wiki/spaces/arc/pages/49512759/Coding+conventions#Use-Smart-Commits%3A

File: MailReporter/Mvc/Controllers/MailReporterController.cs (hunk 1 above)
Comment (AnujAngooral) [reply MailReporter/Mvc/Controllers/MailReporterController.cs]: Removed the extra commented code and updated the ticket id

File: MailReporter/Mvc/Controllers/MailReporterController.cs [hunk 2]
```diff
@@ -53,31 +56,43 @@ public async Task<IActionResult> Mandrill()
...
+        {
+            var req = HttpContext.Request;
+            try
+            {
+                await _sendInBlueService.ProcessEvent(payload);
+            }
+            catch (Exception e)
+            {
                 throw;
             }
```
Comment (bruno-de-andrade) [inline MailReporter/Mvc/Controllers/MailReporterController.cs:97]: I don't understand the reason for the catch and rethrow. Maybe it was for debugging? I think you can safely remove it.

Comment (AnujAngooral) [reply MailReporter/Mvc/Controllers/MailReporterController.cs:97]: I will remove this try-catch and will add the logging in a separate task.

**Review by bruno-de-andrade — APPROVED** [review]

**Review by hm-harshit — APPROVED** [review]

**Pull request merged**